functions/clash_impact.h

```diff
@@ -0,0 +1,35 @@
+#ifndef FUNCTIONS_CLASH_IMPACT_H
+#define FUNCTIONS_CLASH_IMPACT_H
+
+#include "../common/saber_base.h"
+
+// Usage: ClashImpact<MIN, MAX>
+// MIN is minimum value Clash is detected
+// MAX is maximum impact to return 32768
+// Returns 0-32768 based on impact strength of clash
+// returned value: INTEGER
+
+template<class MIN, class MAX>
+class ClashImpactF {
+public:
+  void run(BladeBase* base) {
+    min_cents_.run(base);
+    max_cents_.run(base);
+    float i = SaberBase::GetClashStrength();
+    value_ = clampi32(((i - (min_cents_.getInteger(0) / 100)) / (max_cents_.getInteger(0) / 100)) * 32768, 0, 32768);
```

Comment:
By dividing with 100, we are throwing away the decimals.
Also, we should be dividing by max - min, not just the maximum.

I think what we want is this:

int min = min_cents_.getInteger();
int max = max_cents_.getInteger();
value_ = clampi32((i - min) * 32768 / (max - min), 0, 32768);

Reply:
I've updated, had to modify though as min and max were 100 times larger than i